rust/kernel/types.rs

```diff
@@ -463,3 +463,102 @@ impl<T: LittleEndian + Copy> core::convert::From<T> for LE<T> {
         LE(value.to_le())
     }
 }
+
+/// Specifies that a type is safely readable from byte slices.
+///
+/// Not all types can be safely read from byte slices; examples from
+/// <https://doc.rust-lang.org/reference/behavior-considered-undefined.html> include [`bool`] that
+/// must be either `0` or `1`, and [`char`] that cannot be a surrogate or above [`char::MAX`].
+///
+/// # Safety
+///
+/// Implementers must ensure that the type is made up only of types that can be safely read from
+/// arbitrary byte sequences (e.g., [`u32`], [`u64`], etc.).
```

Comment:
I find this sentence a bit weird, what about
```suggestion
/// Implementers must ensure that any bit pattern is valid for this type.
```

Reply:
Yes, thanks, yours is much better.